### PR TITLE
Feature: Frigate service widget

### PR DESCRIPTION
**Pull request by j0ton**
## Proposed change

Implemented a new [Frigate](https://frigate.video) service widget (Feature Request [#2857](https://github.com/gethomepage/homepage/discussions/2857))

Screenshot 2024-07-15 at 8 34 24 PM

### Example with `enableRecentEvents` option set to `true`:
Screenshot 2024-07-15 at 8 33 54 PM

### Example of event thumbnail preview on hovering over event:
Screenshot 2024-07-15 at 8 35 13 PM

Closes [#2857](https://github.com/gethomepage/homepage/discussions/2857)

## Type of change

- [x] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [x] If applicable, I have added corresponding documentation changes.
- [x] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

**Comment by shamoon** [comment]
Thanks. Please post some example api output for us to test with. And I think the thumbnail part will have to be removed, it really doesn’t comply with our guidelines and we really emphasize consistency

**Comment by narsaw** [comment]
Homepage Dev please consider leaving in the thumbnail feature. ITs a really big deal to have this feature.

**Comment by shamoon** [comment]
Homepage is not designed to be a "Swiss-army knife" single tool to replace everything, the widget features are to get a glance into other tools. The thumbnail should be removed. Presumably those events can link to somewhere else, that would be ok.

**Comment by j0ton** [comment]
> Thanks. Please post some example api output for us to test with. And I think the thumbnail part will have to be removed, it really doesn’t comply with our guidelines and we really emphasize consistency

OK no problem I'll remove the thumbnail and possibly look at linking to the events.
I've attached output for the 2 api calls used: `/api/stats` and `/api/events`
[stats.json](https://github.com/user-attachments/files/16239341/stats.json)
[events.json](https://github.com/user-attachments/files/16239340/events.json)

File: src/widgets/frigate/component.jsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import Container from "components/services/widget/container";
+import Block from "components/services/widget/block";
+import useWidgetAPI from "utils/proxy/use-widget-api";
+
+function Event({ camera, label, startTime, score }) {
+  const { i18n } = useTranslation();
+
+  const dateFormatter = new Intl.DateTimeFormat(i18n.language, { timeStyle: "short", dateStyle: "medium" });
+  const percentFormatter = new Intl.NumberFormat(i18n.language, { style: "percent" });
```
Comment (shamoon) [inline src/widgets/frigate/component.jsx:11]: This should be done either by passing in `t`. But also at this point perhaps just moving the `Event` component inside the `map` below. After all, this is only getting used once and its (now) pretty small.

Anyway, so then itd be e.g. `t("common.percent", { value: score })`

File: src/widgets/frigate/component.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        <Block label="frigate.version" />
+      </Container>
+    );
+  }
+
+  return (
+    <>
+      <Container service={service}>
+        <Block
+          label={data.num_cameras === 1 ? "frigate.camera" : "frigate.cameras"}
```
Comment (shamoon) [inline src/widgets/frigate/component.jsx:56]:
```suggestion
          label="frigate.cameras"
```

File: public/locales/en/common.json [hunk 3]
```diff
@@ -900,5 +900,11 @@
         "open": "Open - US Market",
         "closed": "Closed - US Market",
         "invalidConfiguration": "Invalid Configuration"
+    },
+    "frigate": {
+        "camera": "Camera",
```
Comment (shamoon) [inline public/locales/en/common.json:905]:
```suggestion
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks. couple small things and I still need to test

**Review by shamoon — APPROVED** [review]
Great thanks

Screenshot 2024-07-17 at 7 06 02 AM

**Pull request merged**